### PR TITLE
[IMPROVE] Use SessionId for credential token in SAML request

### PR DESCRIPTION
**Pull request by MohammedEssehemy**
to be more stricter for current session

### CODE REVIEW

**Comment by engelgabriel** [comment]
Thanks @MohammedEssehemy 

Can you please explain what is going to be the difference in the end user experience?

**Comment by MohammedEssehemy** [comment]
@engelgabriel because the credential Token was generated randomly client side, a malicious attacker could send curated link to victim and when the user log this will lead to account take over by attacker.

Now, we use session id and so attacker has no way to change his session id.

**Comment by engelgabriel** [comment]
I understand. Seems like a good idea indeed!

**Comment by MohammedEssehemy** [comment]
any updates here?

File: app/meteor-accounts-saml/server/saml_server.js [hunk 1]
```diff
@@ -92,11 +92,11 @@ Meteor.methods({
 });
 
 Accounts.registerLoginHandler(function(loginRequest) {
-	if (!loginRequest.saml || !loginRequest.credentialToken) {
```
Comment (rafaelks) [inline app/meteor-accounts-saml/server/saml_server.js]: This change broke the mobile apps I think (cc @geekgonecrazy)

Comment (geekgonecrazy) [reply app/meteor-accounts-saml/server/saml_server.js]: Yup I think so as well...  :(

Comment (MohammedEssehemy) [reply app/meteor-accounts-saml/server/saml_server.js]: can't the mobile app send the connection id instead of the random secret?

**Pull request merged**